### DISCUSSION TITLE
'Master profile' to update all other profiles

### DISCUSSION THREAD

**Discussion by OilyHands** [runelite/runelite#16462] [Ideas]
When I discover a new plugin, sometimes I'd like for it to apply to all of my profiles. However, this is not possible. I would have to individually configure the plugin with each profile active.

I suggest a 'master profile' where you could globally configure plugins - for all of your profile. Alternatively, an option on a plugin 'copy configuration to all other profiles'

**Comment by raysdigitalfootprint** [comment]
Being able to set a profile to be a "slave" to another profile would be ideal for a a stretched mode profile and a fixed mode profile when you want the same ground markers ect. I'm not sure how it would work though

**Comment by gxgx55** [comment]
Yes, I'd like this as well. My current situation is that I play two accounts with two profiles, but they only differ by a few settings, so currently I either have to make changes to both of the profiles, or I have to nuke the other profile, make a copy of the one I changed, and then apply the divergent changes to the copy again.

I'd like for my primary profile changes to apply to my secondary profile as well, UNLESS they are already different, while changing secondary profile will diverge those settings, unlinking those particular settings. Does that make sense?

**Comment by allisonkong** [comment]
My suggestion on implementation for this (since I came here to suggest a similar idea) is to make it possible to create a hierarchical profile tree. Each child profile is configured to only customize a set of whitelisted plugins.
- When constructing plugin config, you start with the root config and override it with each child down the tree.
- When plugin config is updated, you walk up the tree until you find the first node that claims that particular plugin. The root profile is a catch-all for all plugins.
- While I expect the common use case to likely have just one root and a handful of child profiles, there's no reason we can't support multiple roots and an arbitrarily deep tree (up to the profile limit).

As an example use case, consider the case where I have a base profile, and a child that overrides entity hider + idle notifier for AFK skilling.
- If I'm on the base profile and make a change to an unrelated plugin, like tile markers or bank tag layouts, it applies to all profiles because they inherit from base.
- If I'm on the child profile and make a change to a not-whitelisted plugin, e.g. recording an item lend/collat in the notes plugin, it gets applied to the base profile (and thus all profiles inherit the change).
- If I'm on the child profile and make a change to entity hider, that change is only applied to the child profile, and base + the other profiles are unaffected.

If this idea sounds reasonable, I am happy to do the legwork to implement it. I'm pretty invested in fixing this, as currently I avoid using profiles altogether because it's too annoying to have to recreate them when I make a change to my "global state" (bank layouts, notes, etc).
